File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        // 정렬하고 실행해야 바로옆 거 비교하면서 진행 가능 
        Arrays.sort(phone_book);
        // System.out.println(Arrays.toString(phone_book));
        for (int i=0; i<phone_book.length-1; i++){
            if (phone_book[i+1].startsWith(phone_book[i])){
                answer = false;
                return answer;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    answer = True
    phone_book.sort()
    for i in range(len(phone_book)-1):
       # for j in range(1,len(phone_book[i+1:])+1):
            if phone_book[i+1].startswith(phone_book[i]) : 
                #print(phone_book[i], phone_book[j])
                answer = False
                break
        #if answer == False:
        #    break
    return answer